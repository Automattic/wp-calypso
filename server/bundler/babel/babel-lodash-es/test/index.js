/** @format */

/**
 * External dependencies
 */
const babel = require( 'babel-core' );

describe( 'babel-lodash-es', () => {
	function transform( code ) {
		return babel.transform( code, {
			plugins: [ require( '..' ) ],
		} ).code;
	}

	test( 'should transform named import from top-level package', () => {
		const code = "import { get } from 'lodash-es'";
		expect( transform( code ) ).toMatchSnapshot();
	} );

	test( 'should transform default import from top-level package', () => {
		const code = "import _ from 'lodash-es'";
		expect( transform( code ) ).toMatchSnapshot();
	} );

	test( 'should transform default import from a submodule', () => {
		const code = "import get from 'lodash-es/get'";
		expect( transform( code ) ).toMatchSnapshot();
	} );

	test( 'should transform aliased default import from a submodule', () => {
		const code = "import theGet from 'lodash-es/get'";
		expect( transform( code ) ).toMatchSnapshot();
	} );

	test( 'should report error on a non-default import from a submodule', () => {
		const code = "import { get } from 'lodash-es/get'";
		expect( () => transform( code ) ).toThrowErrorMatchingSnapshot();
	} );
} );
