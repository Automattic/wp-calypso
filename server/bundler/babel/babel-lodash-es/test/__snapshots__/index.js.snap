// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-lodash-es should report error on a non-default import from a submodule 1`] = `"unknown: babel-lodash-es: Could not transform a non-default import from lodash-es/submodule"`;

exports[`babel-lodash-es should transform aliased default import from a submodule 1`] = `"import { get as theGet } from 'lodash';"`;

exports[`babel-lodash-es should transform default import from a submodule 1`] = `"import { get } from 'lodash';"`;

exports[`babel-lodash-es should transform default import from top-level package 1`] = `"import _ from 'lodash';"`;

exports[`babel-lodash-es should transform named import from top-level package 1`] = `"import { get } from 'lodash';"`;
